File: javascript/utils/download_files.js
import axios from 'axios';
import fs from 'fs';
import path from 'path';

export async function downloadFile(url, outputFolder) {
    const fileName = path.basename(url.split('?')[0]);
    const outputPath = path.join(outputFolder, fileName);

    if (!fs.existsSync(outputFolder)) {
        fs.mkdirSync(outputFolder, { recursive: true });
    }

    const writer = fs.createWriteStream(outputPath);

    const response = await axios({
        url,
        method: 'GET',
        responseType: 'stream'
    });

    response.data.pipe(writer);

    return new Promise((resolve, reject) => {
        writer.on('finish', resolve);
        writer.on('error', reject);
    });
}

File: utils/download_files.py

import os
import aiohttp

async def download_file(url, output_folder):
    if not os.path.exists(output_folder):
        os.makedirs(output_folder)
    filename = url.split('/')[-1]
    filepath = os.path.join(output_folder, filename)
    async with aiohttp.ClientSession() as session:
        async with session.get(url) as response:
            response.raise_for_status()
            with open(filepath, 'wb') as f:
                f.write(await response.read())
